src/d/actor/d_a_leaflift.cpp

```diff
@@ -4,98 +4,352 @@
+        }
+        fVar2 = posOffset.abs();
+        if (!posOffset.normalizeRS() ) {
+          return;
+        }
+        cLib_addCalcAngleS2(&param2->m440, -fVar2 * fVar1, 8, 0x200);
+        fVar1 = cM_ssin(param2->m440);
+        param2->mZeroQuat.x = posOffset.x * fVar1;
+        param2->mZeroQuat.y = posOffset.y * fVar1;
+        param2->mZeroQuat.z = posOffset.z * fVar1;
+        param2->mZeroQuat.w = cM_scos(param2->m440);
+    }
+    param2->m43D = 0;
+    return;
 }
 
 /* 00000CAC-00000D34       .text set_mtx__9daLlift_cFv */
 void daLlift_c::set_mtx() {
-    /* Nonmatching */
+    mpModel->setBaseScale(scale);
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_stack_c::ZXYrotM(current.angle);
+    mpModel->setBaseTRMtx(mDoMtx_stack_c::get());
+    return;
 }
 
 /* 00000D34-00000D9C       .text setMoveBGMtx__9daLlift_cFv */
 void daLlift_c::setMoveBGMtx() {
-    /* Nonmatching */
+    mDoMtx_stack_c::transS(current.pos.x, current.pos.y, current.pos.z);
+    mDoMtx_stack_c::YrotM(current.angle.y);
+    mDoMtx_stack_c::quatM(&mBaseQuat);
+    MTXCopy(mDoMtx_stack_c::get(), mMtx);
+    return;
 }
 
 /* 00000D9C-00000F24       .text _execute__9daLlift_cFv */
 bool daLlift_c::_execute() {
-    /* Nonmatching */
+    Quaternion local_1c;
+    cXyz local_28;
+    float distXZ = fopAcM_searchActorDistanceXZ(this, dComIfGp_getPlayer(0));
+    m444++;
+    if ((m43F) && (m43F--, !m43F)) {
+        m43E = 1;
+    }
+    if ((!m43C) && (m43E)) {
+        mZeroQuat = ZeroQuat;
+        if (distXZ > 270.0f) {
+            m468 = 1;
+      }
+    }
+    MoveDownLift();
+    set_mtx();
+    setMoveBGMtx();
+    m43C = 0;
+    mpModel->calc();
+    mDoMtx_quatSlerp(&mBaseQuat, &mZeroQuat, &local_1c, 0.25f);
+    mBaseQuat = local_1c;
+    mcBgW->Move();
+    local_28 = current.pos;
+    local_28.y -= m_height;
+    mCyl.SetC(local_28);
+    dComIfG_Ccsp()->Set(&mCyl);
+    emitterCtrl();
+    return TRUE;
 }
 
 /* 00000F24-00000FE0       .text emitterCtrl__9daLlift_cFv */
 void daLlift_c::emitterCtrl() {
-    /* Nonmatching */
+    int iVar1;
+  
+    if (m49C == 200) {
+        if (mEmitter1) {
+            mEmitter1->becomeInvalidEmitter();
+            mEmitter1 = NULL;
+        }
+        if (mEmitter2) {
+            mEmitter2->becomeInvalidEmitter();
+            mEmitter2 = NULL;
+        }
+    }
+    if (mEmitter1) {
+        mEmitter1->setGlobalRTMatrix(mBGMtx);
+    }
+    if (mEmitter2) {
+        mEmitter2 ->setGlobalRTMatrix(mBGMtx);
+    }
+    return;
 }
 
 /* 00000FE0-000010E0       .text MoveDownLift__9daLlift_cFv */
-void daLlift_c::MoveDownLift() {
-    /* Nonmatching */
+BOOL daLlift_c::MoveDownLift() {
+    float maxSpeed = m_max_speed;
+    float minSpeed = m_min_speed;
+    if (!m468) {
+      return TRUE;
+    }
+    float fVar4 = cLib_addCalc(&current.pos.y, home.pos.y, 0.1f, maxSpeed, minSpeed);
```

Comment:
Could name the local like this to match the name `upVel` in the other func
```suggestion
    float downVel = cLib_addCalc(&current.pos.y, home.pos.y, 0.1f, maxSpeed, minSpeed);
```